tests/Type/Symfony/data/form_data_type.php

```diff
@@ -29,6 +29,9 @@ class DataClassType extends AbstractType
 
 	public function buildForm(FormBuilderInterface $builder, array $options): void
 	{
+		assertType('GenericFormDataType\DataClass|null', $builder->getData());
+		assertType('GenericFormDataType\DataClass|null', $builder->getForm()->getData());
```

Comment:
Unfortunately this can be `null` for embedded form, cf 
https://github.com/symfony/symfony/issues/5694
https://github.com/symfony/symfony/issues/28741
https://stackoverflow.com/questions/21862168/calling-builder-getdata-from-within-a-nested-form-always-returns-null